lib/gs-category.c

```diff
@@ -536,14 +536,17 @@ gboolean
 gs_category_equal (GsCategory *a,
 		   GsCategory *b)
 {
-	const char *a_id, *b_id;
-
 	g_return_val_if_fail (GS_IS_CATEGORY (a), FALSE);
 	g_return_val_if_fail (GS_IS_CATEGORY (b), FALSE);
 
-	a_id = gs_category_get_id (a);
-	b_id = gs_category_get_id (b);
-	return g_strcmp0 (a_id, b_id) == 0;
+	if (a == b) {
+		return TRUE;
+	} else {
```

Comment:
You don't need, and shouldn't use the else block here (since you return and never reach it otherwise). It's also a common use in g-s, so IMO we should do it.